AllReadyApp/Web-App/AllReady/Views/Campaign/ManageCampaign.cshtml

```diff
@@ -0,0 +1,102 @@
+                </div>
+                <div class="form-group">
+                    <button type="button" class="btn btn-default btn-sm" data-bind="css: { active: campaigns.showOld }, click: campaigns.showOld.toggle">
+                        <i class="fa" data-bind="css: { 'fa-square-o': !campaigns.showOld(), 'fa-check-square-o': campaigns.showOld }"></i>
+                        Include Completed Campaigns
+                    </button>
+                    <span data-bind="text: '(' + total + ')' " ></span>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
+
+
+<div class="row">
+    <div class="hide text-center" data-bind="css: { hide: campaigns.filtered().length }">
+        <div class="alert alert-warning" role="alert">
+            <strong>No matching campaigns.</strong>
+        </div>
+    </div>
+    <div class="col-md-12">
+        <div id="campaignGrid" class="table-responsive">
+            <table class="table" data-bind="css: { hide: !campaigns.filtered().length }">
+                <tr>
+                    <th class="col-md-3"><span title="Name of the campaign">Title</span></th>
+                    <th class="col-md-3"><span title="Organizer of the campaign">Organizer</span></th>
+                    <th class="col-md-4"><span title="Description of the campaign">Description</span></th>
+                    <th class="col-md-2"><span title="Date of the campaign">Dates</span></th>
+                </tr>
+                <!-- ko foreach: campaigns.filtered -->
+                <tr>
+                    <td>
+                        <a data-bind="attr: { href: '/Campaign/' + Id }, text: Name"></a>
+                        <i data-bind="visible: Featured" class="fa fa-star star-highlight" title="Featured" aria-hidden="true"></i>
+                    </td>
+                    <td>
+                        <span data-bind="text: ManagingOrganizationName"></span>
+                        <div data-bind="visible:ManagingOrganizationLogo != '' || ManagingOrganizationLogo != null"><br/><img width="250" data-bind="attr:{src: ManagingOrganizationLogo}" /></div>
+                    </td>
+                    <td>
+                        <span data-bind="text: Description"></span>
+                    </td>
+                    <td>
+                        <span data-bind="text: displayDate()"></span>
+                    </td>
+                </tr>
+                <!-- /ko -->
+            </table>
+
+            <div data-bind="css: { hide: !featuredCampaigns }">
+                <i class="fa fa-star star-highlight" aria-hidden="true"></i>
+                Featured campaign(s)
+            </div>
+        </div>
+    </div>
+</div>
+
+@section scripts {
+    <script>
+        var modelCampaigns = @Json.Serialize(Model);
+    </script>
+    <script src="~/js/campaigns.js"></script>
+}
```

Comment:
Looking at this view as a whole, and also taking into consideration that it directly relies on C:\Projects\allReady\AllReadyApp\Web-App\AllReady\wwwroot\js\campaigns.js in order to function, why do we need to return an `AllReady.ViewModels.Campaign.CampaignViewModel` from `AuthorizedCampaignsQueryHandler` in order for this page to work?

Try this. List out all the properties you need in order for this page/ko code to run. If there are any un-used properties from the CampaignViewModel in the .js/cshtml files, then you should not re-use CampaignViewModel. Instead, create a new model, optimized for consumption by this view/ko code. Only query for what you need in `AuthorizedCampaignsQueryHandler` and only pass back a model to be used by this  view/ko code.

Reusing views is not really a good practice if you're querying for, building a model, and then passing back a model that is only using 15-20% of the properties that the view (and in this case, ko code) rely on.

This is especially true in this project b/c some of the ViewModels are so out of control. CampaignViewModel has 25 properties on it and also contains logic in some of those properties as well as logic in the constructor that takes `Model.Campaign`.

I can't see how a developer is supposed to know which constructor to use in what circumstance when constructing this ViewModel